File: Model/src/main/java/org/xdi/oxauth/model/common/Parameters.java
/*
 * oxAuth is available under the MIT License (2008). See http://opensource.org/licenses/MIT for full text.
 *
 * Copyright (c) 2014, Gluu
 */

package org.xdi.oxauth.model.common;

/**
 * @author Yuriy Zabrovarnyy
 * @version 0.9, 20/12/2012
 */

public enum Parameters implements HasParamName {
    SESSION_ID("session_id"),
    REQUEST_SESSION_ID("request_session_id");

    private String paramName;
    private String nameToAppend;

    private Parameters(String p_paramName) {
        paramName = p_paramName;
        nameToAppend = "&" + p_paramName + "=";
    }

    public String getParamName() {
        return paramName;
    }

    public String nameToAppend() {
        return nameToAppend;
    }
}

File: Model/src/main/php/org/xdi/oxauth/model/common/Parameters.php
<?php
namespace org\xdi\oxauth\model\common;

/**
 * Description of Parameters
 *
 * @author Gabin Dongmo
 * @version August 9, 2017
 */
class Parameters implements HasParamName {
    
    /**
     * Session Id
     */
    const SESSION_ID = "session_id";
    
    /**
     * Request Session Id
     */
    const REQUEST_SESSION_ID = "request_session_id";
    
    /**
     * Parameter Name
     * @var String 
     */
    private $paramName;
    
    /**
     * Name to append
     * @var String 
     */
    private $nameToAppend;
    
    /**
     * Constructor
     * @param String Parameter
     */
    public function __construct($parameter = "") {
        $this->paramName = $parameter;
        $this->nameToAppend = "&" + $this->paramName + "=";
    }
    
    /**
     * Gets the parameter name entered
     * @return String Parameter name
     */
    public function getParamName() {
        return $this->paramName;
    }
    
    /**
     * Get the paramater appended
     * @return String Name to append
     */
    public function nameToAppend(){
        return $this->nameToAppend;
    }
}



?>
